diff --git a/arch/arm/mach-aspeed/include/mach/ast2400_platform.h b/arch/arm/mach-aspeed/include/mach/ast2400_platform.h
index fad9bbc..b7c5cfe 100644
--- a/arch/arm/mach-aspeed/include/mach/ast2400_platform.h
+++ b/arch/arm/mach-aspeed/include/mach/ast2400_platform.h
@@ -99,11 +99,11 @@
 
 #define AST_LPC_BRIDGE						0x60000000
 
-#define ASR_VIDEO_MEM_SIZE				0x2800000		//40MB
-#define ASR_VIDEO_MEM 					(AST_DRAM_BASE + (SZ_8M*10))//(AST_DRAM_BASE + SZ_256M)
+#define AST_VIDEO_MEM_SIZE				0x2800000		//40MB
+#define AST_VIDEO_MEM 					(AST_DRAM_BASE + (SZ_8M*10))//(AST_DRAM_BASE + SZ_256M)
 
 #define AST_CRT0_MEM_SIZE				SZ_8M
-#define AST_CRT0_MEM_BASE				(ASR_VIDEO_MEM + ASR_VIDEO_MEM_SIZE)
+#define AST_CRT0_MEM_BASE				(AST_VIDEO_MEM + AST_VIDEO_MEM_SIZE)
 
 #define AST_CURSOR0_MEM_SIZE			SZ_1M
 #define AST_CURSOR0_MEM_BASE			(AST_CRT0_MEM_BASE + AST_CRT0_MEM_SIZE)
diff --git a/arch/arm/plat-aspeed/dev-fb.c b/arch/arm/plat-aspeed/dev-fb.c
index 6725edb..3da0c46 100644
--- a/arch/arm/plat-aspeed/dev-fb.c
+++ b/arch/arm/plat-aspeed/dev-fb.c
@@ -191,7 +191,7 @@ void __init ast_add_device_fb(void)
 	ast_scu_multi_func_crt();
 	ast_scu_init_crt();
 	
-	fb_plat_data.clock_src = ast_get_clk_source();
+//	fb_plat_data.clock_src = ast_get_clk_source();
 
 	platform_device_register(&ast_fb0_device);
 #ifdef AST_CRT1_BASE	
