diff --git a/drivers/net/ethernet/faraday/ftgmac100.c b/drivers/net/ethernet/faraday/ftgmac100.c
index bc87879..6956915 100644
--- a/drivers/net/ethernet/faraday/ftgmac100.c
+++ b/drivers/net/ethernet/faraday/ftgmac100.c
@@ -2216,7 +2216,8 @@ void ncsi_start(struct net_device *dev) {
         //TODO: This is an issue in  Get_Version_ID that always returns
         //mezz_type to be -1, so it only calls Get_MAC_Address_bcm.
         //It may need to work with Mlx to find a solution.
-#if defined(CONFIG_FBY2) || defined(CONFIG_YOSEMITE)    //For multi-host NIC initialization
+// #if defined(CONFIG_FBY2) || defined(CONFIG_YOSEMITE)    //For multi-host NIC initialization
+#if defined(CONFIG_FBY2)    //For multi-host NIC initialization
         // Try Mlx first
         Get_MAC_Address_mlx(dev);
         Set_MAC_Affinity_mlx(dev);
@@ -2243,7 +2244,8 @@ void ncsi_start(struct net_device *dev) {
           Get_MAC_Address_bcm(dev);
           mdelay(500);
         } else {
-          Get_MAC_Address_intel(dev);
+          // Get_MAC_Address_intel(dev);
+          Get_MAC_Address_bcm(dev);
           mdelay(500);
         }
 
