diff --git a/arch/arm/plat-aspeed/dev-i2c.c b/arch/arm/plat-aspeed/dev-i2c.c
index 36892b4..05e3f11 100644
--- a/arch/arm/plat-aspeed/dev-i2c.c
+++ b/arch/arm/plat-aspeed/dev-i2c.c
@@ -318,6 +318,18 @@ struct platform_device ast_i2c_dev6_device = {
 	.resource = ast_i2c_dev6_resources,
 	.num_resources = ARRAY_SIZE(ast_i2c_dev6_resources),
 };
+
+struct platform_device ast_i2c_dev6_device_1M = {
+	.name = "ast-i2c",
+	.id = 6,
+	.dev = {
+		.dma_mask = &ast_i2c_dma_mask,
+		.coherent_dma_mask = 0xffffffff,
+		.platform_data = &ast_i2c_data_1M,
+	},
+	.resource = ast_i2c_dev6_resources,
+	.num_resources = ARRAY_SIZE(ast_i2c_dev6_resources),
+};
 #endif
 
 #if defined(AST_I2C_DEV7_BASE)
@@ -899,7 +911,10 @@ static struct i2c_board_info ast_i2c_board_info_2[] __initdata = {
 
 //Under I2C Dev 3
 static struct i2c_board_info ast_i2c_board_info_3[] __initdata = {
-  // Slot#1 IPMB interface
+  //
+  // {
+  //   I2C_BOARD_INFO("24c02", 0x52),
+  // },
 };
 
 //Under I2C Dev 4
@@ -925,28 +940,28 @@ static struct i2c_board_info ast_i2c_board_info_7[] __initdata = {
 //Under I2C Dev 8
 static struct i2c_board_info ast_i2c_board_info_8[] __initdata = {
   // FRUID
-  {
-    I2C_BOARD_INFO("24c64", 0x51),
-  },
+  // {
+  //   I2C_BOARD_INFO("24c64", 0x51),
+  // },
 };
 
 //Under I2C Dev 9
 static struct i2c_board_info ast_i2c_board_info_9[] __initdata = {
   // Inlet and Outlet temp. sensors
-  {
-    I2C_BOARD_INFO("tmp421", 0x4e),
-  },
-  {
-    I2C_BOARD_INFO("tmp421", 0x4f),
-  },
+  // {
+  //   I2C_BOARD_INFO("tmp421", 0x4e),
+  // },
+  // {
+  //   I2C_BOARD_INFO("tmp421", 0x4f),
+  // },
 };
 
 //Under I2C Dev 10
 static struct i2c_board_info ast_i2c_board_info_10[] __initdata = {
   // Hotswap Sensor
-  {
-    I2C_BOARD_INFO("adm1278", 0x40),
-  },
+  // {
+  //   I2C_BOARD_INFO("adm1278", 0x40),
+  // },
 };
 
 //Under I2C Dev 11
@@ -960,9 +975,9 @@ static struct i2c_board_info ast_i2c_board_info_11[] __initdata = {
 //Under I2C Dev 12
 static struct i2c_board_info ast_i2c_board_info_12[] __initdata = {
   // Mezz Card Mezz_SMB bus (FRUID, GPIO expander, QSFP+)
-  {
-    I2C_BOARD_INFO("24c64", 0x51),
-  },
+  // {
+  //   I2C_BOARD_INFO("24c64", 0x51),
+  // },
 };
 
 /* end of CONFIG_YOSEMITE */
@@ -1266,46 +1281,46 @@ void __init ast_add_device_i2c(void)
   ast_i2c_data_1M.reg_gr = ast_i2c_data.reg_gr;// 1MHz reg_gr setting
 
 	platform_device_register(&ast_i2c_dev0_device);
-	platform_device_register(&ast_i2c_dev1_device_1M);
+	platform_device_register(&ast_i2c_dev1_device);
 	platform_device_register(&ast_i2c_dev2_device);
-	platform_device_register(&ast_i2c_dev3_device_1M);
+	platform_device_register(&ast_i2c_dev3_device);
 	platform_device_register(&ast_i2c_dev4_device);
 	platform_device_register(&ast_i2c_dev5_device_1M);
-	platform_device_register(&ast_i2c_dev6_device);
+	platform_device_register(&ast_i2c_dev6_device_1M);
 	platform_device_register(&ast_i2c_dev7_device_1M);
 	platform_device_register(&ast_i2c_dev8_device);
-	platform_device_register(&ast_i2c_dev9_device);
-	platform_device_register(&ast_i2c_dev10_device);
-	platform_device_register(&ast_i2c_dev11_device);
-	platform_device_register(&ast_i2c_dev12_device);
-	platform_device_register(&ast_i2c_dev13_device);
+	// platform_device_register(&ast_i2c_dev9_device);
+	// platform_device_register(&ast_i2c_dev10_device);
+	// platform_device_register(&ast_i2c_dev11_device);
+	// platform_device_register(&ast_i2c_dev12_device);
+	// platform_device_register(&ast_i2c_dev13_device);
 
 	i2c_register_board_info(0, ast_i2c_board_info_0,
-		              ARRAY_SIZE(ast_i2c_board_info_0));
+		            ARRAY_SIZE(ast_i2c_board_info_0));
 	i2c_register_board_info(1, ast_i2c_board_info_1,
-                  ARRAY_SIZE(ast_i2c_board_info_1));
+                    ARRAY_SIZE(ast_i2c_board_info_1));
 	i2c_register_board_info(2, ast_i2c_board_info_2,
-                  ARRAY_SIZE(ast_i2c_board_info_2));
+                    ARRAY_SIZE(ast_i2c_board_info_2));
 	i2c_register_board_info(3, ast_i2c_board_info_3,
-                  ARRAY_SIZE(ast_i2c_board_info_3));
+                    ARRAY_SIZE(ast_i2c_board_info_3));
 	i2c_register_board_info(4, ast_i2c_board_info_4,
-                  ARRAY_SIZE(ast_i2c_board_info_4));
+                    ARRAY_SIZE(ast_i2c_board_info_4));
 	i2c_register_board_info(5, ast_i2c_board_info_5,
-                  ARRAY_SIZE(ast_i2c_board_info_5));
+                    ARRAY_SIZE(ast_i2c_board_info_5));
 	i2c_register_board_info(6, ast_i2c_board_info_6,
-                  ARRAY_SIZE(ast_i2c_board_info_6));
+                    ARRAY_SIZE(ast_i2c_board_info_6));
 	i2c_register_board_info(7, ast_i2c_board_info_7,
-                  ARRAY_SIZE(ast_i2c_board_info_7));
+                    ARRAY_SIZE(ast_i2c_board_info_7));
 	i2c_register_board_info(8, ast_i2c_board_info_8,
-		              ARRAY_SIZE(ast_i2c_board_info_8));
-	i2c_register_board_info(9, ast_i2c_board_info_9,
-                  ARRAY_SIZE(ast_i2c_board_info_9));
-	i2c_register_board_info(10, ast_i2c_board_info_10,
-                  ARRAY_SIZE(ast_i2c_board_info_10));
-	i2c_register_board_info(11, ast_i2c_board_info_11,
-                  ARRAY_SIZE(ast_i2c_board_info_11));
-	i2c_register_board_info(12, ast_i2c_board_info_12,
-                  ARRAY_SIZE(ast_i2c_board_info_12));
+		            ARRAY_SIZE(ast_i2c_board_info_8));
+	// i2c_register_board_info(9, ast_i2c_board_info_9,
+ //                  ARRAY_SIZE(ast_i2c_board_info_9));
+	// i2c_register_board_info(10, ast_i2c_board_info_10,
+ //                  ARRAY_SIZE(ast_i2c_board_info_10));
+	// i2c_register_board_info(11, ast_i2c_board_info_11,
+ //                  ARRAY_SIZE(ast_i2c_board_info_11));
+	// i2c_register_board_info(12, ast_i2c_board_info_12,
+ //                  ARRAY_SIZE(ast_i2c_board_info_12));
 }
 
 #elif defined(CONFIG_LIGHTNING)
